fud2/fud-core/src/run.rs

```diff
@@ -104,7 +118,10 @@ pub struct OpEmitData {
     /// The commands run whenever the op run. Each of these must be run in order in a context where
     /// the variables in each cmd are supplied. In particular, this means that variables of the
     /// form "$[i|o]<digit>" are in scope.
-    pub cmds: Vec<String>,
+    ///
+    /// The second and third element of the tuple is a list of the indecies of other commands the given
+    /// command depends on.
+    pub cmds: Vec<ShellCmd>,
```

Comment:
My suggestion on the other PR to make this a single thing instead of a `Vec` of course no longer applies! We want a list here.